smac_planner/src/a_star.cpp

```diff
@@ -411,7 +411,7 @@ AStarAlgorithm<NodeSE2>::NodePtr AStarAlgorithm<NodeSE2>::getAnalyticPath(
   prev = node;
   for (const auto & node_pose : possible_nodes) {
     const auto & n = node_pose.first;
-    if (!n->wasVisited()) {
+    if (!n->wasVisited() and n->getIndex()!=_goal->getIndex()) {
```

Comment:
```suggestion
    if (!n->wasVisited() && n->getIndex() != _goal->getIndex()) {
```

Just to be consistent in styling